### PR TITLE
Crowdfunding pallet

### PR DESCRIPTION
**Pull request by f-gate**
As per: https://github.com/ImbueNetwork/imbue/issues/111

- [x] Move Project setup into new pallet
- [x] Write new tests for pallet-crowdfunding (up to proposal conversion)
- [x] Fix up benchmarks
- [x] Remove the pesky pallet-timestamp
- [x] Migration for contribution
- [ ] Discuss improvements

### CODE REVIEW

**Comment by mshankarrao** [comment]
> As per: #111
> 
> * [x]  Move Project setup into new pallet
> * [x]  Write new tests for pallet-crowdfunding (up to proposal conversion)
> * [ ]  Fix up benchmarks
> * [x]  Remove the pesky pallet-timestamp
> * [x]  Migration for contribution
> * [ ]  Discuss improvements

so are they asking to create new pallet for project?

File: pallets/crowdfunding/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+            CrowdFundsInRound::<T>::remove(key, RoundType::ContributionRound);
+            weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+        });
+        
+        weight
+    }
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn new_crowdfund(
```
Comment (samelamin) [inline pallets/crowdfunding/src/lib.rs:357]: Once the deposits pr is merged, we will need to add it here too

File: pallets/crowdfunding/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+            agreement_hash,
+            crowdfund_key,
+            required_funds,
+            currency_id,
+        ));
+
+        Ok(crowdfund_key)
+    }
+
+    pub fn try_update_existing_crowdfund(
```
Comment (samelamin) [inline pallets/crowdfunding/src/lib.rs:405]: this can never fair right? if so just call it `update_crowdfund`

Comment (f-gate) [reply pallets/crowdfunding/src/lib.rs]: done

**Pull request merged**